src/plugins/data/public/query/saved_query/saved_query_service.test.ts

```diff
@@ -287,6 +292,175 @@ describe('saved query service', () => {
         ])
       );
     });
+
+    it('should correctly parse a json query string', async () => {
+      mockSavedObjectsClient.get.mockReturnValue({
+        id: 'foo',
+        attributes: {
+          title: 'foo',
+          description: 'bar',
+          query: {
+            language: 'kuery',
+            query:
+              '{ "email": [{"type": "work","address":"work@***.com" }, {"type": "home", "address":"home@***.com"}]}',
+          },
+        },
+      });
+
+      const response = await getSavedQuery('foo');
+      expect(response.attributes.query.query).toEqual({
+        email: [
+          { type: 'work', address: 'work@***.com' },
+          { type: 'home', address: 'home@***.com' },
+        ],
+      });
+    });
+
+    it('should correctly parse a json object', async () => {
+      mockSavedObjectsClient.get.mockReturnValue({
+        id: 'foo',
+        attributes: {
+          title: 'foo',
+          description: 'bar',
+          query: {
+            language: 'kuery',
+            query: {
+              email: [
+                { type: 'work', address: 'work@***.com' },
+                { type: 'home', address: 'home@***.com' },
+              ],
+            },
+          },
+        },
+      });
+
+      const response = await getSavedQuery('foo');
+      expect(response.attributes.query.query).toEqual({
+        email: [
+          { type: 'work', address: 'work@***.com' },
+          { type: 'home', address: 'home@***.com' },
+        ],
+      });
+    });
+
+    it('should handle null string with single quote', async () => {
+      mockSavedObjectsClient.get.mockReturnValue({
+        id: 'foo',
+        attributes: {
+          title: 'foo',
+          description: 'bar',
+          query: {
+            language: 'kuery',
+            query: 'null',
+          },
+        },
+      });
+
+      const response = await getSavedQuery('foo');
+      expect(response.attributes.query.query).toEqual('null');
+    });
+
+    it('should handle null string with double quote', async () => {
+      mockSavedObjectsClient.get.mockReturnValue({
+        id: 'foo',
+        attributes: {
+          title: 'foo',
+          description: 'bar',
+          query: {
+            language: 'kuery',
+            query: 'null',
```

Comment:
This test seems to be the same as the previous test. Also are these null tests really necessary? seems redundant to have 4 separate tests just to test 2 variations of null and undefined, especially when the original issue had no issue with null or undefined queries but rather with how quotes were escaped.

Reply:
For unit tests on both null and undefined, yes I think it is not necessary to test both edges cases. I will remove tests on undefined.

Yes, the original issue is not about null/undefined. But I think previous getSavedQuery unit test does not cover all query cases 1) a json string which will be parsed to object 2) a regular string with single/double quote which will be parsed to string 3) some edge cases: null, undefined

Since I am changing getSavedQuery to cover all the cases, would like to add tests even the original issue is not considering json string or edge cases.